github/github.go

```diff
@@ -427,10 +443,18 @@ func CheckResponse(r *http.Response) error {
 	if err == nil && data != nil {
 		json.Unmarshal(data, errorResponse)
 	}
-	if r.StatusCode == http.StatusUnauthorized && strings.HasPrefix(r.Header.Get(headerOTP), "required") {
+	switch {
+	case r.StatusCode == http.StatusUnauthorized && strings.HasPrefix(r.Header.Get(headerOTP), "required"):
 		return (*TwoFactorAuthError)(errorResponse)
+	case r.StatusCode == http.StatusForbidden && r.Header.Get(headerRateRemaining) == "0" && strings.HasPrefix(errorResponse.Message, "API rate limit exceeded for "):
+		return &RateLimitError{
+			Rate:     parseRate(r),
```

Comment:
Technically, it's possible to avoid having to parse rate from `*http.Response` here. It's already done previously in `Do` when it does:

```Go
response := newResponse(resp) // response.Rate is parsed and available here.

c.rateMu.Lock()
c.rate = response.Rate
c.rateMu.Unlock()

err = CheckResponse(resp) // CheckResponse is not given the Rate, so it has to parse it from resp a second time...
```

The problem is that `func CheckResponse(r *http.Response) error` is exported and its signature does not accept the rate.

However, it might be possible to create an unexported `func checkResponseWithRate(r *http.Response, rate Rate) error` and use that in `Do` instead.